kvdb-rocksdb/src/lib.rs

```diff
@@ -162,35 +162,36 @@ pub struct DatabaseConfig {
 	/// write buffer size for each column including the default one.
 	/// If the memory budget of a column is not specified,
 	/// `DB_DEFAULT_COLUMN_MEMORY_BUDGET_MB` is used for that column.
-	pub memory_budget: HashMap<Option<u32>, MiB>,
+	pub memory_budget: HashMap<u32, MiB>,
 	/// Compaction profile.
 	pub compaction: CompactionProfile,
 	/// Set number of columns.
-	pub columns: Option<u32>,
+	pub columns: NonZeroU32,
```

Comment:
So this is problematic. Over in `parity-ethereum` there's e.g. a constant like so:

```
/// The migration from v10 to v11.
/// Adds a column for node info.
pub const TO_V11: ChangeColumns = ChangeColumns {
	pre_columns: Some(6),
	post_columns: Some(7),
	version: 11,
};
```

Translating that to the new API the above must be:

```
pub const TO_V11: ChangeColumns = ChangeColumns {
	pre_columns: NonZeroU32::new(7),
	post_columns: NonZeroU32::new(8),
	version: 11,
};
```

Note how I can't unwrap the `NonZeroU32`s because it's not a const fn, so now the `ChangeColumns` type has to use `Option<NonZeroU32>` which sort of snowballs and leads to quite a lot of noise. I think that is a problem in general with the `NonZero*` types: using them is rather clumsy and I wonder if their usefulness actually carries its weight.
How about we switch to runtime validation instead and take a `u32` straight up here instead? At the end of the day the database is not user-configurable and tbh I prefer a few sharp edges to noisy code.

/cc @ordian

Reply:
@dvdplm Something like [this commit](https://github.com/paritytech/parity-common/pull/278/commits/74b703e1d41d80e26614d98b3e523f07d248f17d)? Already pushed to this PR branch.

Reply:
Yeah, that's pretty much exactly what I had here too. Still would like to hear @ordian's thoughts. :)

Reply:
Well, I [used to have a strong opinion on that](https://github.com/paritytech/parity-common/pull/140#discussion_r282538527), but now I think it doesn't have to be so invasive. We can have `NonZeroU32` only at the API boundary and convert it internally from u32 somewhere.
But I'm fine with assert as well.